Tests/Common/Securities/Futures/FuturesExpiryFunctionsTests.cs

```diff
@@ -16,12 +16,114 @@
+            public SymbolData(String symbol, List<Dates> list)
+            {
+                this.symbol = symbol;
+                this.dateList = list;
+            }
+        }
+
+        [Test]
+        public void TestAllExpiryDateFunctions()
+        {
+            var _path = Directory.GetCurrentDirectory();
+            _path = _path.Substring(0,_path.Length-10) + "\\Common\\Securities\\Futures\\FuturesExpiryFunctionsTestData.xml";
+            IList<String> symbolsForNineThirtyEasternTime = new List<String>
+            {
+                QuantConnect.Securities.Futures.Indices.NASDAQ100EMini,
+                QuantConnect.Securities.Futures.Indices.SP500EMini,
+                QuantConnect.Securities.Futures.Indices.Dow30EMini
+            };
+            IList<String> symbolsForNineSixteenCentralTime = new List<String>
+            {
+                QuantConnect.Securities.Futures.Currencies.GBP,
+                QuantConnect.Securities.Futures.Currencies.CAD,
+                QuantConnect.Securities.Futures.Currencies.JPY,
+                QuantConnect.Securities.Futures.Currencies.CHF,
+                QuantConnect.Securities.Futures.Currencies.EUR,
+                QuantConnect.Securities.Futures.Currencies.AUD,
+                QuantConnect.Securities.Futures.Currencies.NZD
+            };
+            IList<String> symbolsForTwelveOne = new List<String>
+            {
+                QuantConnect.Securities.Futures.Financials.Y30TreasuryBond,
+                QuantConnect.Securities.Futures.Financials.Y10TreasuryNote,
+                QuantConnect.Securities.Futures.Financials.Y5TreasuryNote,
+                QuantConnect.Securities.Futures.Financials.Y2TreasuryNote
+            };
+            IList<String> symbolsForTwelveOclock = new List<String>
+            {
+                QuantConnect.Securities.Futures.Meats.LiveCattle,
+                QuantConnect.Securities.Futures.Meats.LeanHogs
+            };
+            using (XmlReader reader = XmlReader.Create(_path))
+            {
+                List<SymbolData> data = new List<SymbolData>();
+                XmlSerializer serializer = new XmlSerializer(typeof(List<SymbolData>));
+                data = (List<SymbolData>)serializer.Deserialize(reader);
+                foreach(var _symdata in data)
+                {
+                    var symbol = _symdata.symbol;
+                    foreach (var dates in _symdata.dateList)
+                    {
+                        var _security = Symbol.CreateFuture(symbol, Market.USA, dates.contractMonth);
+                        var func = FuturesExpiryFunctions.FuturesExpiryFunction(_security.ID.Symbol);
+                        var calculated = func(_security.ID.Date);
+                        var expected = dates.lastTrade;
+                        if (symbolsForNineThirtyEasternTime.Contains(_security.ID.Symbol))
+                        {
+                            expected = expected + new TimeSpan(13,30,0);
```

Comment:
You may consider using the Nunit `[TestCase]` attribute (http://nunit.org/?p=testCase&r=2.5) instead of calculating what the expected value of the test should be. It may make it easier to understand what the expected value of a test is if a test is failing.